velox/type/TimestampConversion.cpp

```diff
@@ -498,14 +571,32 @@ int64_t fromDateString(const char* str, size_t len) {
   int64_t daysSinceEpoch;
   size_t pos = 0;
 
-  if (!tryParseDateString(str, len, pos, daysSinceEpoch, true)) {
+  if (!tryParseDateString(str, len, pos, daysSinceEpoch, ParseMode::kStrict)) {
     VELOX_USER_FAIL(
         "Unable to parse date value: \"{}\", expected format is (YYYY-MM-DD)",
         std::string(str, len));
   }
   return daysSinceEpoch;
 }
 
+int32_t
+castFromDateString(const char* str, size_t len, bool isNonStandardCast) {
+  int64_t daysSinceEpoch;
+  size_t pos = 0;
+
+  auto mode = isNonStandardCast ? ParseMode::kNonStandardCast
+                                : ParseMode::kStandardCast;
+  if (!tryParseDateString(str, len, pos, daysSinceEpoch, mode)) {
+    VELOX_USER_FAIL(
+        "Unable to parse date value: \"{}\"."
+        "In Spark SQL, valid date string patterns include "
```

Comment:
Thank you for updating the error message. Just a small nit: would it make sense to have different error messages based on the value of isNonStandardCast. This would ensure the error message accurately conveys what pattern it expects since castFromDateString might be used in different paths with different  isNonStandardCast values.
Also, this would mean we dont need to mention presto and spark explicitly in the error message and keep it generic.